### ISSUE TITLE
Roadmap for 5.5

### ISSUE THREAD

**Issue by Beep6581** [Beep6581/RawTherapee#4429]
Issues in the 5.5 milestone:
https://github.com/Beep6581/RawTherapee/milestone/5

**Comment by Beep6581** [comment]
There are currently 28 open issues pending for the 5.5 release. We need to review them to set a release date for 5.5 so that it does not take forever.
@agriggio @Desmis @Floessie @heckflosse @Hombre57 @iliasg @Thanatomanic could you please go through all of them and see if you can help complete any of them? If you can, please assign yourself to the issue.

**Comment by agriggio** [comment]
@Beep6581 I'll be on vacation from tomorrow until the end of the month. camera and a couple of lenses, but no laptop... I'll check GitHub on the phone, but that's about it...

**Comment by Beep6581** [comment]
Let's aim for a feature-freeze this weekend (November 10), then a release candidate about two weeks later (November 24), and if all goes well, then a stable release some time around December 17.

**Comment by Beep6581** [comment]
You can review the `release-5.5` branch here:
https://github.com/Beep6581/RawTherapee/compare/release-5.5

Release notes:
https://github.com/Beep6581/RawTherapee/blob/release-5.5/RELEASE_NOTES.txt

I will tag 5.5-rc1 tomorrow.

**Comment by Hombre57** [comment]
@Beep6581 Please add commit e2490e5381ce97cc1d77034b65ea15cd1c73dc1c to RT5.5.

**Comment by agriggio** [comment]
Also 7cfd9dc8f4269a2bf774385275161839e708fd77 should go in

**Comment by Kildor** [comment]
What is last day for translations? Do I have a week?
I have an update for Russian translation, but I want to improve it a little before 5.5

**Comment by Beep6581** [comment]
@Kildor RC1 is today, final is a week before Christmas (~17th).

**Comment by Beep6581** [comment]
@aferrero2707 @gaaned92 @benitoite [5.5-rc1](https://github.com/Beep6581/RawTherapee/releases/tag/5.5-rc1) is tagged, please link me to binaries when available (manual, CI, etc).

**Comment by Benitoite** [comment]
Here is a build for MacOS 10.9+

https://filebin.net/toy5x0dwtx4p764p/RawTherapee_OSX_10.9_64_5.5-rc1.zip?t=anwhtlgi

```
Version: 5.5-rc1
Branch: 5.5-rc1
Commit: ca6724557
Commit date: 2018-12-03
Compiler: clang 10.0.0.10001145
Processor: generic x86
System: Apple
Bit depth: 64 bits
Gtkmm: V3.22.3
Lensfun: V0.3.95.0
Build type: release
Build flags: -std=c++11 -std=c++11 -mtune=generic -Werror=unused-label -mmacosx-version-min=10.9 -flto -Xpreprocessor -fopenmp /opt/local/lib/libiomp5.dylib -I/opt/local/include -Werror=unknown-pragmas -Wall -Wno-unused-result -Wno-deprecated-declarations -O3 -DNDEBUG -ftree-vectorize
Link flags:  -mtune=generic -headerpad_max_install_names -flto
OpenMP support: ON
MMAP support: ON

```

Lensfun Timestamp= `1536512901`

Build by Richard Barber
:barber:

**Comment by aferrero2707** [comment]
AppImage package: [RawTherapee-releases-5.5-rc1-20181203.AppImage](https://github.com/Beep6581/RawTherapee/releases/download/nightly/RawTherapee-releases-5.5-rc1-20181203.AppImage)

The automated Win64 build will come in a short while...

**Comment by aferrero2707** [comment]
Win64 continuous integration:
- portable: [RawTherapee_releases-5.5-rc1.zip](https://github.com/aferrero2707/rt-win64/releases/download/continuous/RawTherapee_releases-5.5-rc1.zip)
- installer: [RawTherapee_releases-5.5-rc1_WinVista_64.exe](https://github.com/aferrero2707/rt-win64/releases/download/continuous/RawTherapee_releases-5.5-rc1_WinVista_64.exe)

**Comment by gaaned92** [comment]
> RawTherapee_releases_5.5-rc1_WinVista_64.zip                                                                                                                    

uploaded at 
https://keybase.pub/gaaned92/RTW64NightlyBuilds/

Lensfun Timestamp= 1536512901

**Comment by gaaned92** [comment]
> RawTherapee_releases_5.5-rc1_WinVista_32.zip  

uploaded at 
https://keybase.pub/gaaned92/RTW32NightlyBuilds/

**Comment by sguyader** [comment]
So far no issues for me with 5.5-rc1.
"Color Regions" should be added to the Color Toning tool tip.

**Comment by Beep6581** [comment]
@gaaned92 Is there currently any benefit to using your build as compared to @aferrero2707 's automated installer? To keep things simple I will only upload one build, and would prefer to use automated ones for obvious reasons, assuming they're as good as your self-made ones.

**Comment by Beep6581** [comment]
@Benitoite I always wondered what that little Christmas stick was below your posts, found out now that it's a `:barber:` hehe :)

**Comment by gaaned92** [comment]
@Beep6581 
As I already wrote, except some minor issues, the automatic build is quite usable. So 5.5RC1 seems an occasion  to use this build and be sure it is ok for final release.
I had no time to update the list of minor issues I sent by PM to @aferrero2707 . I will do that before we.

**Comment by Benitoite** [comment]
There are some interesting differences between the builds.

| Build | @aferrero2707  |@gaaned92  |  |  |
|---:|---|---|---|---|
|   |none  |`-Wno-aggressive-loop-optimizations`  |  |  |  |
|   |none  |`-mfpmath=sse `   |  |  |  |
| Opt. level  | `-O2` |  `-O3` |   |  |  |
|  compiler | `gcc` 7.3.0 | `gcc` 8.2.1 |  | | |
| type | `RelWithDebInfo`  | `release` | | | |
| libbz2| | ✔️  | | | 
| libdatrie | |✔️ | | | 
| libfftw3 | | ✔️  | | |
| libfftw3f | ✔️ | ✔️  | | |
| libfribidi | | ✔️ | | |
| libgraphite2 | | ✔️ | | |
| gtk+ | 3.22.26 | 3.22.30 | | |
| libiconv | | ✔️ | | | |
| libgtk-win32 | ✔️ | | | |
| libHalf | ✔️ | | | |
| liblex, etc | ✔️ | | | |
| libjasper | ✔️ | | | |
| libzstd | | ✔️ | | |
| libtre | | ✔️  | | |
| libthai | | ✔️ | | |
| libsystre | | ✔️ | | |
| libssp | ✔️ | | | |
| libquadmath | ✔️ | | | |
| libpugixml | ✔️ | | | |



Also, @aferrero2707 the autobuild's AboutThisBuild.txt does not have the Windows line endings:
![capture](https://user-images.githubusercontent.com/17919239/49482011-48bfd080-f7e2-11e8-8e8a-53d1c9a86e1a.PNG)

**Comment by Beep6581** [comment]
Builds are now live.
https://rawtherapee.com/blog/rawtherapee-5.5-release-candidate-1

**Comment by Benitoite** [comment]
@Beep6581 Here is a new RC2 build for mac...
https://kd6kxr.keybase.pub/RawTherapee_OSX_10.9_64_5.5-rc2.zip

```
Version: 5.5-rc2
Branch: 5.5-rc2
Commit: d4ba923ff
Commit date: 2018-12-11
Compiler: clang 10.0.0.10001145
Processor: generic x86
System: Apple
Bit depth: 64 bits
Gtkmm: V3.22.3
Lensfun: V0.3.95.0
Build type: release
Build flags: -std=c++11 -std=c++11 -mtune=generic -Werror=unused-label -mmacosx-version-min=10.9 -flto -Wall -Wuninitialized -Wno-deprecated-declarations -Wno-unused-result -Xpreprocessor -fopenmp /opt/local/lib/libiomp5.dylib -I/opt/local/include -Werror=unknown-pragmas -O3 -DNDEBUG -ftree-vectorize
Link flags:  -mtune=generic -headerpad_max_install_names -flto
OpenMP support: ON
MMAP support: ON
```

:barber:

**Comment by Beep6581** [comment]
RawTherapee 5.5 is tagged \o/
Well done and thank you everyone!

**Comment by Beep6581** [comment]
@aferrero2707 @Benitoite @gaaned92 please link me to the builds when ready, then I will make public the release announcement.

**Comment by Benitoite** [comment]
@Beep6581 
Here is a macOS 10.9/10/11/12/13/14 build of `5.5`
https://kd6kxr.keybase.pub/RawTherapee_OSX_10.9_64_5.5.zip

```

Version: 5.5
Branch: 5.5
Commit: d9ac23555
Commit date: 2018-12-17
Compiler: clang 10.0.0.10001145
Processor: generic x86
System: Apple
Bit depth: 64 bits
Gtkmm: V3.22.3
Lensfun: V0.3.95.0
Build type: release
Build flags: -std=c++11 -std=c++11 -mtune=generic -Werror=unused-label -mmacosx-version-min=10.9 -flto -Wall -Wuninitialized -Wno-deprecated-declarations -Wno-unused-result -Xpreprocessor -fopenmp /opt/local/lib/libiomp5.dylib -I/opt/local/include -Werror=unknown-pragmas -O3 -DNDEBUG -ftree-vectorize
Link flags:  -mtune=generic -headerpad_max_install_names -flto
OpenMP support: ON
MMAP support: ON

```

🌮 🎉

**Comment by gaaned92** [comment]
@Beep6581 

     RawTherapee_5.5_WinVista_64.zip

    uploaded at

https://keybase.pub/gaaned92/RTW64NightlyBuilds


```
Version: 5.5
Branch: 5.5
Commit: d9ac23555
Commit date: 2018-12-17
Compiler: gcc 8.2.1
Processor: generic x86
System: Windows
Bit depth: 64 bits
Gtkmm: V3.22.3
Lensfun: V0.3.2.0
Build type: release
Build flags:  -m64 -mwin32 -msse2 -mfpmath=sse -mthreads -Wno-aggressive-loop-optimizations -Wno-parentheses -std=c++11 -mtune=generic -Werror=unused-label -Wall -Wuninitialized -Wno-deprecated-declarations -Wno-unused-result -fopenmp -Werror=unknown-pragmas  -DNDEBUG  -O3 -ftree-vectorize
Link flags: -m64  -mthreads  -static-libgcc   -mtune=generic  -s  -O3  -fno-use-linker-plugin
OpenMP support: ON
MMAP support: ON
```

**Comment by gaaned92** [comment]
and for the uncompromising lovers of W32:

     RawTherapee_5.5_WinVista_32.zip

    uploaded at
https://keybase.pub/gaaned92/RTW32NightlyBuilds

**Comment by aferrero2707** [comment]
@Beep6581 AppImage package: https://github.com/Beep6581/RawTherapee/releases/download/nightly/RawTherapee-releases-5.5-20181217.AppImage

**Comment by Floessie** [comment]
For the fun of it I built yesterday's `dev` on [Debian 10 ppc32](https://filebin.net/67o9xqyr6ow94z65/rt_ppc.png). PowerPC is demanding (and unique) for being both big endian and `char == unsigned char`.

No related complaints from the compiler, binary works just fine.

Don't get fooled by "generic x86" in the screenshot: There is no `-mtune=generic` for PowerPC, so I modified the ProcTarget. Maybe we could add a really generic ProcTarget, or can it be omitted when calling `cmake`?

Best,
Flössie
